src/System.Text.Primitives/System/Text/InvariantParser_int.cs

```diff
@@ -98,10 +88,18 @@ unsafe public static bool TryParse(byte* utf8Text, int index, int length, out in
             value = 0;
             bytesConsumed = 0;
             bool negative = false;
+            bool signed = false;
 
             if (utf8Text[index] == '-')
             {
                 negative = true;
+                signed = true;
```

Comment:
Can't you remove this `signed` flag and just not increment bytesConsumed in these two cases?